lang/syn/src/idl/defined.rs

```diff
@@ -496,7 +496,11 @@ pub fn gen_idl_type(
                 use crate::parser::context::CrateContext;
                 use quote::ToTokens;
 
-                let source_path = proc_macro2::Span::call_site().source_file().path();
+                let source_path = match proc_macro2::Span::call_site().local_file() {
+                    Some(source_path) => source_path,
+                    None => return Err(anyhow!("No source path found")),
```

Comment:
I think returning error here could cause undesired behavior because we always want [the rest of the code](https://github.com/solana-foundation/anchor/blob/649b9d6c5ef19d8e15ea35888f0b1a846cd03b48/lang/syn/src/idl/defined.rs#L577-L615) to run.

The old method didn't return an optional path, so it's probably unlikely for the new `local_file` method to return `None` in real world usage, but it's better to be safe than sorry here.

Reply:
I see. I'm not quite sure how you'd want me to avoid indenting [the stuff that's conditional on the path existing](https://github.com/solana-foundation/anchor/blob/649b9d6c5ef19d8e15ea35888f0b1a846cd03b48/lang/syn/src/idl/defined.rs#L500-L574) in that case :sweat_smile: . Do you have a suggestion? Feel free to push to the branch yourself if that's simpler.

Reply:
gotcha - will change it to an empty path instead